### PR TITLE
Add XCharts data viz

### PR DESCRIPTION
**Pull request by mohammedhamdhan**

### CODE REVIEW

**Review by matthewyeo1 — COMMENTED** [review]
Clean way of letting users see their expenditure distribution in a vibrant manner!

**Review by matthewyeo1 — COMMENTED** [review]
Remember to add relevant JUnit tests for this feature

File: src/main/java/seedu/duke/summary/ExpenseClassifier.java [hunk 1]
```diff
@@ -1,16 +1,82 @@
...
 
 public class ExpenseClassifier {
+    private Map<Categories, List<String>> categoryKeywords;
+
+    public ExpenseClassifier() {
+        initializeCategoryKeywords();
+    }
 
-    public ExpenseClassifier() {}
+    private void initializeCategoryKeywords() {
```
Comment (NandhithaShree) [inline src/main/java/seedu/duke/summary/ExpenseClassifier.java:18]: I love how keywords automatically categorise expenses. Very impressive!

File: src/main/java/seedu/duke/commands/ExpenseCommand.java [hunk 2]
```diff
@@ -471,26 +486,94 @@ private void showCategorySummary() {
...
+     */
+    private void showPieChart(List<String> categoryNames, List<Number> categoryValues) {
+        // Create Chart
+        PieChart chart = new PieChartBuilder()
+                .width(800)
+                .height(600)
+                .title("Expenses by Category")
+                .theme(ChartTheme.GGPlot2)
+                .build();
+
```
Comment (nandhananm7) [inline src/main/java/seedu/duke/commands/ExpenseCommand.java:538]: Good job implementing pie charts to improve user experience

File: src/main/java/seedu/duke/summary/ExpenseClassifier.java [hunk 3]
```diff
@@ -1,16 +1,82 @@
...
+                "shoes", "jewelry", "home decor", "furniture", "beauty", "cosmetics", "boutique", "outlet"));
+        categoryKeywords.put(Categories.Travel, Arrays.asList(
+                "flight", "hotel", "transport", "train", "taxi",
+                "bus", "cruise", "rental car", "itinerary", "tour", "excursion", "travel agent", "reservation",
+                "airport", "visa", "backpacker"));
+        categoryKeywords.put(Categories.Entertainment, Arrays.asList(
+                "movie", "concert", "game", "party", "music",
+                "theater", "comedy", "festival", "show", "art", "exhibition", "opera", "dance", "amusement",
+                 "performance", "arcade"));
+    }
```
Comment (nandhananm7) [inline src/main/java/seedu/duke/summary/ExpenseClassifier.java:34]: Very useful feature for users to improve user-friendliness

File: src/test/java/seedu/duke/commands/ExpenseCommands.java [hunk 4]
```diff
@@ -296,6 +302,169 @@ void testDisplayUnsettledExpensesZeroExpenses() {
...
+    void testShowCategorySummary() {
+        // Add expenses with more explicit category indicators
+        budgetManager.addExpense(new Expense("Lunch", "restaurant food", "01-01-2025", 25.50));
+        budgetManager.addExpense(new Expense("T-shirt", "clothes shopping", "02-01-2025", 35.00));
+        budgetManager.addExpense(new Expense("Movie ticket", "entertainment movie cinema", "03-01-2025", 15.00));
+
+        // First provideInput with answer for visualization prompt ('n' for no)
+        provideInput("n\n");
+
+        // Run the category summary
```
Comment (nandhananm7) [inline src/test/java/seedu/duke/commands/ExpenseCommands.java:316]: Good job testing for all the new additions and edge cases

File: src/main/java/seedu/duke/commands/ExpenseCommand.java [hunk 5]
```diff
@@ -390,7 +405,7 @@ public void showExpenseSummary() {
         System.out.println("Choose summary view:");
         System.out.println("1. Monthly Summary");
         System.out.println("2. Category-wise Summary");
-        System.out.println("3. Back to main menu");
+        System.out.println("3. Cancel");
```
Comment (Ashertan256) [inline src/main/java/seedu/duke/commands/ExpenseCommand.java:408]: good for making ti clearer for the user!

File: src/test/java/seedu/duke/commands/ExpenseCommands.java [hunk 6]
```diff
@@ -296,6 +302,169 @@ void testDisplayUnsettledExpensesZeroExpenses() {
...
+        // Get the output and print it for debugging
+        String output = outContent.toString();
+        System.err.println("CATEGORY SUMMARY OUTPUT:\n" + output);
+        
+        // Verify the output contains the summary title
+        assertTrue(output.contains("Category-wise Expense Summary"), 
+                   "Output should contain the summary title");
+        
+        // More comprehensive checks for food-related terms
+        assertTrue(
```
Comment (Ashertan256) [inline src/test/java/seedu/duke/commands/ExpenseCommands.java:328]: Detailed assertions, good!
